kre-py/src/compression.py

```diff
@@ -95,11 +95,13 @@ def compress_if_needed(
     out = compress(data)
 
     if len(out) > max_size:
-        raise Exception(
-            f"compressed message {size_in_kb(out)} exceeds "
-            f"maximum size allowed of {bytes_to_kb(max_size)}."
-        )
+        raise Exception("compressed message exceeds maximum size allowed")
 
-    logger.info("Original message size: %s. Compressed: %s", size_in_kb(data), size_in_kb(out))
+    logger.debug("compressed message exceeds maximum size allowed: current" +
+                f"message size {bytes_to_mb(data)}MB, max allowed size {bytes_to_mb(out)}MB")
```

Comment:
Left this `Info` log message as is for message size information, and move the new debug message to above the line where the `Exception` is raised, here it will never be executed as the `Exception` is raised first.

```suggestion
    logger.info("Original message size: %s. Compressed: %s", size_in_kb(data), size_in_kb(out))
```